dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -282,7 +282,7 @@ internal KernelFunction(string name, string? pluginName, string description, IRe
         Verify.NotNull(kernel);
 
         using var activity = s_activitySource.StartActivity(this.Name);
-        ILogger logger = kernel.LoggerFactory.CreateLogger(this.Name) ?? NullLogger.Instance;
+        ILogger logger = kernel.LoggerFactory.CreateLogger($"Microsoft.SemanticKernel.KernelFunction.{this.Name}") ?? NullLogger.Instance;
```

Comment:
Suggest using the current function type istead + the name.
```suggestion
        ILogger logger = kernel.LoggerFactory.CreateLogger($"{this.GetType()}.{this.Name}") ?? NullLogger.Instance;
```

This will create something like:

- `Microsoft.SemanticKernel.PromptFunction.NameOfPromptFUnction`
- `Microsoft.SemanticKernel.MethodFunction.NameOfMethodFunction`